proposals/0002.md

```diff
@@ -0,0 +1,318 @@
+* Title: Structured error handling in Rust
+* Champion: @CodeSandwich
+* Conversation (optional): [thread on radicle.community](https://radicle.community/t/proposal-structured-error-handling-in-rust/179)
+* References:
+  * [The previous error handling proposal](https://github.com/radicle-dev/radicle-decisions/blob/master/proposals/0001.md)
+  * [`std::error::Error`](https://doc.rust-lang.org/std/error/trait.Error.html)
+  * [A discussion about the good design of errors](https://users.rust-lang.org/t/do-people-not-care-about-printable-error-chains-a-k-a-how-to-nicely-implement-display-for-an-error/35362)
+
+# Summary/Motivation/Introduction
+We need good error handling to make our lives easier.
+We've already agreed to use `thiserror` and `anyhow` in the
+[previous error handling proposal](https://github.com/radicle-dev/radicle-decisions/blob/master/proposals/0001.md).
+They are fantastic tools, but we're still lacking a common design language.
+There are many ways to create an error and some of them don't play well with others.
+Without common conventions handling errors is problematic too.
```

Comment:
It seems like these sentences were meant to be "bullet points"? They are all in different lines but when parsed they are all inlined.

Reply:
That's intended. It's easier to work on a document cleanly split into sentences and lines at the same time.